src/app/data-explorer/querybuilder/querybuilder.component.ts

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2022 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+/*
+Query builder source: https://github.com/zebzhao/Angular-QueryBuilder
+*/
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { QueryBuilderConfig } from 'angular2-query-builder';
+import { DataType } from '@maurodatamapper/mdm-resources';
+import { DataElementSearchResult } from 'src/app/data-explorer/data-explorer.types';
+
+@Component({
+  selector: 'mdm-querybuilder',
+  templateUrl: './querybuilder.component.html',
+  styleUrls: ['./querybuilder.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryBuilderComponent {
+  // Query Builder Configuration Begin
+  config: QueryBuilderConfig = {
+    fields: {},
+  };
+
+  operatorMap: { [key: string]: string[] } = {
+    /* eslint-disable-next-line */
+    string: ['=', '!=', 'contains', 'like', 'startswith', 'endswith'],
+  };
+
+  allowRuleset = true;
+  allowCollapse = true;
+  persistValueOnFieldChange = false;
+
+  query = {
+    condition: 'and',
+    rules: [],
+  };
+  // Query Builder Configuration End
+
+  private _title = 'Query Builder';
+  private showQueryBuilderText: string = 'Show ' + this._title + ' Editor';
+  private hideQueryBuilderText: string = 'Hide ' + this._title + ' Editor';
+  private showQueryBuilderIcon = 'arrow_drop_down';
+  private hideQueryBuilderIcon = 'arrow_drop_up';
+
+  @Input() set dataElements(value: DataElementSearchResult[]) {
+    this.setupConfig(value);
+  }
+  @Input() set title(value: string) {
+    this._title = value;
+    this.showQueryBuilderText = 'Show ' + this._title + ' Editor';
+    this.hideQueryBuilderText = 'Hide ' + this._title + ' Editor';
+    this.setToggleQueryBuilderText();
```

Comment:
I had to leave the input being more involved because the button needs to be refreshed when the button is set. If this is simplified as suggested, then the button ends up showing the default title until it is clicked.
`  private _title = 'Query Builder';
...
  @Input() set title(value: string) {
    this._title = value;
    this.setToggleQueryBuilderText();
  }`